crates/ty_python_semantic/src/types/call/bind.rs

```diff
@@ -2762,6 +2763,51 @@ impl<'a, 'db> ArgumentTypeChecker<'a, 'db> {
             .or(self.signature.return_ty)
             .zip(self.call_expression_tcx.annotation);
 
+        let tcx_specialization = self
+            .call_expression_tcx
+            .annotation
+            .and_then(|annotation| annotation.class_specialization(self.db));
+
+        let promote_literals = |typevar: GenericContextTypeVar<'db>, ty: Type<'db>| -> Type<'db> {
+            let bound_typevar = typevar.bound_typevar();
+
+            if typevar.is_inherited() && bound_typevar.variance(self.db).is_invariant() {
```

Comment:
What does `is_inherited` mean here? Why does it matter whether a typevar is inherited or not?

(It would be great if `is_inherited` could have a doc-comment explaining what it does 😃)

Reply:
Ah, I think I see -- you're using the term to mean that the typevar is associated with a class's generic context rather than the generic context of a function or method? So it doesn't have any thing to do with the generic context of a class's base classes etc.?